[Baizhou Cai] iP

Windofbitter opened the pull request:
#WindBro 
> He who every morning plans the transactions of that day and follows that plan carries a thread that will guide him through the labyrinth of the most busy life. - Victor Hugo [source](https://blog.rescuetime.com/time-management-quotes/) 

WindBro helps you to plan your tasks by setting deadline and remember things. 

- text-based 
- visualization 
- ~~easy~~ _VERY EASY_ to use

All you need to do it: 
1. Download it [here](https://github.com/Windofbitter/ip/releases/tag/A-Jar)
2. Open it
3. Talk to Wind 
4. Let it manages tasks for you

It is **FREE**! 

Features:
- [x] manage tasks
- [x] set deadline for deadline task
- [x] set start date and end date for event task
- [x] A gui chatbot

If you are a java programmer, you can practice by looking at the source code, here is the   `start` method of the program to start the gui:
```
  @Override
    public void start(Stage stage) {
        try {
            FXMLLoader fxmlLoader = new FXMLLoader(Main.class.getResource("/view/MainWindow.fxml"));
            AnchorPane ap = fxmlLoader.load();
            Scene scene = new Scene(ap);
            stage.setScene(scene);
            fxmlLoader.<MainWindow>getController().setDuke(duke);
            stage.show();
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Windofbitter committed:
Finished

Windofbitter committed:
finished

Windofbitter committed:
Finished

Windofbitter committed:
Finished

Windofbitter committed:
"finished"

Windofbitter committed:
"finished"

Windofbitter committed:
finished

Windofbitter committed:
"finished"

Windofbitter committed:
"finished"

Windofbitter committed:
"finished"

Windofbitter committed:
"finished"

Windofbitter committed:
"finished"

Windofbitter committed:
"finished"

Windofbitter committed:
finished

Windofbitter committed:
finished

Windofbitter committed:
finished

Windofbitter committed:
Merge remote-tracking branch 'upstream1/add-gradle-support'
To to the IP

Windofbitter committed:
add gradle

Windofbitter committed:
finish JUnit Test

Windofbitter committed:
fixed bugs

Windofbitter committed:
Add JavaDoc

Windofbitter committed:
Modify Coding Standard

Windofbitter committed:
Add find command

Windofbitter committed:
Merge branch 'branch-A-JavaDoc'
Merge java doc level

Windofbitter committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/command/Command.java
#	src/main/java/duke/command/ListCommand.java
#	src/main/java/duke/command/MarkCommand.java
#	src/main/java/duke/command/UnmarkCommand.java
#	src/main/java/duke/common/TaskList.java
#	src/main/java/duke/exception/DukeException.java
#	src/main/java/duke/exception/InvalidInputException.java
#	src/main/java/duke/exception/StorageOperationException.java
#	src/main/java/duke/exception/UnknownCommandException.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/ui/Ui.java

Windofbitter committed:
optimized import

Windofbitter committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/ui/Ui.java

Windofbitter committed:
Add checkstyle

Windofbitter committed:
Add GUI

kailashgautham commented on src/main/java/seedu/duke/gui/Launcher.java:
```diff
@@ -0,0 +1,9 @@
+
+import javafx.application.Application;
+
+public class Launcher {
```
Hi, I really like how clean your code is! Perhaps you could add Javadocs for this class to explain its purpose?

kailashgautham commented on src/main/java/seedu/duke/command/CheckCommand.java:
```diff
@@ -0,0 +1,63 @@
+            }
+        }
+
+
```
Hey, I was wondering if you needed two empty lines here?

kailashgautham commented on src/main/java/seedu/duke/command/EventCommand.java:
```diff
@@ -0,0 +1,57 @@
+
+
+/**
+ * Represents an event command initiated by the user. <code>EventCommand</code> would create an event task
```
Would it be better to split this javadoc into two lines?

kailashgautham commented on src/main/java/seedu/duke/storage/TaskEncoder.java:
```diff
@@ -0,0 +1,30 @@
+     * @param task the Task object to be encoded
+     * @return The encoded string for a task
+     */
+    public static String encodeTask(Task task) {
```
I really like the way you have done the encoding of tasks into a string!

kailashgautham reviewed (commented):
Your codebase looks good! I noticed a few of the JavaFX classes don't have Javadocs, so you could consider working on that. Overall, well done!

Windofbitter committed:
Add assertions

To ensure Duke is properly injected to MainWindow class.
To ensure that we are decoding the correct Task

Add an assertion on Duke that is not null when displaying initializing message
Add an assertion on Task to be an instance of event when decoding it.

Assertions help to identify potential bugs in programs.

Windofbitter committed:
Improve code quality

It helps to improve readability and easier to debug

Remove a scanner field in the ui class as it never used and the method readCommand as it is no longer needed
Restructure the layout to separate codes performing different structures

By doing this, it adheres to the code quality guide provided by the course

Windofbitter committed:
Merge pull request #1 from Windofbitter/A-Assertions

Add assertions

Windofbitter committed:
Merge pull request #2 from Windofbitter/branch-A-CodeQuality

Improve code quality

Windofbitter committed:
Add ci actions

It helps to utilize github actions to perform CI

A .yml file is added

This is required to perform CI

Windofbitter committed:
Merge branch 'master' of https://github.com/Windofbitter/ip

Windofbitter committed:
Add detect duplicate extensions

This extension gives Duke more functionality

I added codes to check whether there are duplicate task whenever a new task is created

Benson15912 commented on src/main/java/seedu/duke/storage/TaskDecoder.java:
```diff
@@ -0,0 +1,85 @@
+/**
+ * The TaskDecoder class implements a method that would decode string and parse it into Task object
+ */
+public class TaskDecoder {
```
Hey BaiZhou! I like the way you implemented this TaskDecoder for your save/load purposes! Clean and neat which really made it easy to understand

Benson15912 commented on src/main/java/seedu/duke/task/Task.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public class Task {
+    private String description;
+    private boolean hasDone;
```
Hmm probably can rename this to isDone? just a little grammatical error

Benson15912 commented on src/main/java/seedu/duke/parser/Parser.java:
```diff
@@ -0,0 +1,212 @@
+ * The Parser class implements a method that would parse the user input into Command
+ */
+public class Parser {
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
```
Maybe can add comments on what these are suppose to represent? Just a suggestion! cause i dont understand the basic commandformat and one arg command format

Benson15912 reviewed (commented):
Looks good to me! Java coding standard is really good and code is really neat and tidy!

Windofbitter committed:
Add some comments

Windofbitter committed:
Merge branch 'branch-C-DetectDuplicates'

Windofbitter committed:
Add some comments

Windofbitter committed:
Add Ui picture and User guide

Windofbitter committed:
Change storage data file path

Windofbitter committed:
Improve code quality

Windofbitter committed:
Remove dead code

Windofbitter committed:
Improve code style

Windofbitter committed:
Correct Ui png name

Windofbitter committed:
Add more information on read me page

Windofbitter committed:
Change gradle jar output name

Windofbitter committed:
Update read me page
